src/qt/rpcconsole.cpp

```diff
@@ -563,6 +563,17 @@ bool RPCConsole::eventFilter(QObject* obj, QEvent *event)
 void RPCConsole::setClientModel(ClientModel *model)
 {
     clientModel = model;
+
+    bool wallet_enabled{false};
+#ifdef ENABLE_WALLET
+    wallet_enabled = WalletModel::isWalletEnabled();
+#endif // ENABLE_WALLET
+    if (model && !wallet_enabled) {
```

Comment:
In commit "Add warning messages to the debug window" (593ba696fb32da558091ac02ad87c4893db4ce97)

Instead of repeating the wallet check from [`BitcoinGUI::BitcoinGUI`](https://github.com/bitcoin/bitcoin/blob/0221420d1a0550cd849e0f3a5ada3738d5931bdd/src/qt/bitcoingui.cpp#L100-L113), I think it'd be less fragile and repetitive to add a simpler `RPCConsole::enableAlerts()` method here:

```c++
void RPCConsole::enableAlerts()
{
    connect(model, &ClientModel::alertsChanged, this, &RPCConsole::updateAlerts);
    updateAlerts(model->getStatusBarWarnings());
}
```

and call it from [`BitcoinGUI::BitcoinGUI`](https://github.com/bitcoin/bitcoin/blob/0221420d1a0550cd849e0f3a5ada3738d5931bdd/src/qt/bitcoingui.cpp#L100-L113)

Reply:
This approach will not work because the `ClientModel` object has not yet been created at that moment. So I'd prefer to leave it as is.